src/components/Navigation/components/Item/Item.tsx

```diff
@@ -263,6 +251,37 @@ export class BaseItem extends React.Component<CombinedProps, State> {
     );
   }
 
+  private getBadgeMarkup() {
```

Comment:
I would really rather avoid these kinds of methods. I know people like them because we are taught to separate out logic, but in React, separation should either be with a component, or not at all. In this case, I would opt for none at all.